### PR TITLE
Box.com OAuth2

### PR DESCRIPTION
**Pull request by aaronalbright**
## Changes

* Updates Tauri to v1.4
* Adds "Sign in to Box" to the WelcomeModal instead of Start
* Full Oauth2 sign in flow to authenticate with Box.com
* New Rust commands
  * get_auth_url - Returns Box sign-in URL to open in new window
  * request_token - Uses authorization code from sign-in to request an API access token and puts  the tokens in the settings store
  * refresh_token - Retrieves refresh token from settings store and requests a new API access token 
 * Includes demo API functions and buttons to test that the token and refresh token are working

## Additional Changes
* Transparent Titlebar component and configuration for a seamless UI window.
* Disable scrolling on body to avoid awkward bouncing from touch inputs
* Added `.oauth` store
* Refactor rust commands to allow for more `hash::` functions
* Added `DEV_` files and `// DEV` comments where needed for PR testing or in-dev features
* Added `pages` folder and re-worked Vite config to simplify build and folder structure. Updated relative URLs accordingly.

### CODE REVIEW

File: package.json
Comment (aaronalbright) [inline package.json:1]: Version bump here cause needed some features from 1.3. For some reason it says 1.4.0 is available for the CLI but wasn't published at thetime.

File: src/oauth/main.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import { RecoilRoot } from 'recoil';
+import OathConfirm from './OathConfirm';
+
+import '../main.css';
+
+ReactDOM.createRoot(window.document.getElementById('root')).render(
+  <React.StrictMode>
+    <RecoilRoot>
```
[the remark this answers is not in the document]
Comment (aaronalbright) [reply src/oauth/main.jsx:10]: Nope! That was old when I thought I'd need to read atoms to. As long as we're just reading/writing directly to the store as it stands, we don't need it.

**Comment by brizandrew** [comment]
Can you update this PR with our new codebase structure and our new DEV_ rule for development-only files.

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -1,74 +1,97 @@
...
+        context.package_info(),
+        &Env::default(),
+        "Artisan",
+        Some(BaseDirectory::Home),
+    )
+    .expect("failed to resolve path");
+
+    let default_settings = HashMap::from([
+        ("aws-id".to_string(), "".into()),
+        ("aws-secret".to_string(), "".into()),
```
Comment (brizandrew) [inline src-tauri/src/main.rs:38]: Let's remove this, and any other mention to these settings from out codebase.

File: src/atoms/auth/read.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * Gets the user's auth config
+ * @function
+ * @returns {Object}
+ */
+export default async function readSettings() {
+  const settingsEntries = await store.auth.get();
+  return Object.fromEntries(settingsEntries);
+}
```
Comment (brizandrew) [inline src/atoms/auth/read.js:11]:
```suggestion
export default async function readAuth() {
  const authEntries = await store.auth.get();
  return Object.fromEntries(authEntries);
}
```

File: src/atoms/selectors/isConfigured.js [hunk 4]
```diff
@@ -10,8 +10,8 @@ const KEY = 'isConfigured';
 const isConfigured = selector({
   key: KEY,
   get: ({ get }) => {
-    const settings = get(settingsAtom);
-    return (!!settings['aws-id'] && !!settings['aws-secret']);
+    const auth = get(authAtom);
+    return (!!Object.keys(auth.box_tokens).length);
```
Comment (brizandrew) [inline src/atoms/selectors/isConfigured.js:14]:
```suggestion
    return (Object.keys(auth.box_tokens).length > 0);
```

File: src/components/App/index.jsx [hunk 5]
```diff
@@ -1,16 +1,47 @@
...
+        setAuth({
+          ...auth,
+          user_id: id,
+          username,
+        });
+      }
+    };
+
+    getUserData().catch(console.error);
+  }, [auth]);
```
Comment (brizandrew) [inline src/components/App/index.jsx:40]: Can we put this in a hook somewhere else? Like how we're handling `useActivateTime`? Just in the interest of keeping this file clean.

File: src/components/SettingsForm/index.jsx [hunk 6]
```diff
@@ -63,14 +52,14 @@ export default function SettingsForm({ isWelcome = false }) {
...
       <Button
-        className={cls(margin.mt4, {
+        className={cls(margin.mt6, {
           [type.textXl]: isWelcome,
           [padding.px8]: isWelcome,
         })}
         variant="solid"
         onClick={handleClick}
       >
-        {isWelcome ? 'Start' : 'Save'}
```
Comment (brizandrew) [inline src/components/SettingsForm/index.jsx:73]: Can we add a Sign Out button which resets the Auth? Should be side by side and only active if isWelcome is not true.

File: src/constants/types.js [hunk 7]
```diff
@@ -39,6 +41,10 @@ export const TYPE_SETTINGS_STORE_KEYS = {
   'preferred-port': string(),
   'aws-id': string(),
   'aws-secret': string(),
```
Comment (brizandrew) [inline src/constants/types.js:43]: Let's remove these

File: src/constants/types.js [hunk 8]
```diff
@@ -39,6 +41,10 @@ export const TYPE_SETTINGS_STORE_KEYS = {
...
+  box_tokens: voidable(object({
+    access_token: voidable(string()), // These are voidable just for testing
+    refresh_token: voidable(string()),
+  })),
```
Comment (brizandrew) [inline src/constants/types.js:47]: These shouldn't be here.

**Pull request merged**